wallet/src/main/scala/org/bitcoins/wallet/WalletStorage.scala

```diff
@@ -0,0 +1,231 @@
+package org.bitcoins.wallet
+
+import ujson._
+
+import scala.collection.JavaConverters._
+import org.bitcoins.core.util.BitcoinSLogger
+import org.bitcoins.core.crypto.AesPassword
+import java.nio.file.Files
+import org.bitcoins.core.crypto.MnemonicCode
+import scodec.bits.ByteVector
+import org.bitcoins.core.crypto.AesEncryptedData
+import org.bitcoins.core.crypto.AesSalt
+import scala.util.Failure
+import scala.util.Success
+import java.nio.file.Paths
+import java.nio.file.Path
+import scala.util.Try
+import org.bitcoins.db.AppConfig
+
+// what do we do if seed exists? error if they aren't equal?
+object WalletStorage extends BitcoinSLogger {
+
+  private[wallet] val ENCRYPTED_SEED_FILE_NAME: String =
+    "encrypted_bitcoin-s_seed.json"
+
+  private object MnemonicJsonKeys {
+    val IV = "iv"
+    val CIPHER_TEXT = "cipherText"
+    val SALT = "salt"
+  }
+
+  /**
+    * Writes the encrypted mnemonic to disk.
+    * If we encounter a file in the place we're about
+    * to write to, we move it to a backup location
+    * with the current epoch timestamp as part of
+    * the file name.
+    */
+  def writeMnemonicToDisk(mnemonic: EncryptedMnemonic)(
+      implicit config: AppConfig): Path = {
+    import mnemonic.{value => encrypted}
+
+    val jsObject = {
+      import MnemonicJsonKeys._
+      Obj(
+        IV -> encrypted.iv.toHex,
+        CIPHER_TEXT -> encrypted.cipherText.toHex,
+        SALT -> encrypted.salt.value.toHex
+      )
+    }
+
+    val path = config.datadir.resolve(ENCRYPTED_SEED_FILE_NAME)
+
+    logger.debug(s"Writing mnemonic to $path")
+
+    val writtenJs = write(jsObject)
```

Comment:
I would prefer if we used things like `ujson.write` rather than importing everything in the `ujson` package object. It makes it easier to read code as `write` is very generic.